backend/server.js

```diff
@@ -16,9 +21,191 @@ const app = express();
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
-app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+// Middleware set up for error msg like DB down.
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next();
+  } else {
+    res.status(503).json({ error: 'Service unavailable' });
+  }
+});
+
+// ------- User schema and model
+const UserSchema = new mongoose.Schema({
+  username: {
+    type: String,
+    required: true,
+    unique: true
+  },
+  password: {
+    type: String,
+    required: true
+  },
+  // npm install crypto, import
+  accessToken: {
+    type: String,
+    default: () => crypto.randomBytes(128).toString('hex')
+  }
+});
+
+const User = mongoose.model('User', UserSchema);
+
+// ----------- TED Talk model for content -------
+const TedTalk = mongoose.model('TedTalk', {
```

Comment:
First you do the user schema and then the user model, in the tedtalk you combine schema and model. Maybe you could have been more consistent and doing the same approach on both.